samples-aspnetcore-3x/self-hosted-login/okta-aspnetcore-mvc-example/Controllers/AccountController.cs

```diff
@@ -1,43 +1,49 @@
 ﻿using Microsoft.AspNetCore.Authentication;
 using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
 using Okta.AspNetCore;
+using Okta.Idx.Sdk;
+using Okta.Idx.Sdk.Configuration;
+using okta_aspnetcore_mvc_example.Models;
+using okta_aspnetcore_mvc_example.Okta;
+using System.Text;
+using System.Threading.Tasks;
 
 namespace okta_aspnetcore_mvc_example.Controllers
 {
     public class AccountController : Controller
     {
-        public IActionResult SignIn()
+        private readonly IIdxClient idxClient;
+
+        public AccountController(IIdxClient idxClient)
         {
-            return View();
+            this.idxClient = idxClient;
         }
 
-        [HttpPost]
-        [ValidateAntiForgeryToken]
-        public IActionResult SignIn([FromForm]string sessionToken)
+        public async Task<IActionResult> SignIn()
         {
-            if (!HttpContext.User.Identity.IsAuthenticated)
-            {
-                var properties = new AuthenticationProperties();
-                properties.Items.Add("sessionToken", sessionToken);
-                properties.RedirectUri = "/Home/";
+            IIdxContext idxContext = await this.idxClient.InteractAsync();
+            string idxContextJson = JsonConvert.SerializeObject(idxContext);
+            HttpContext.Session.SetString(idxContext.State, idxContextJson);
```

Comment:
Curious, can you do `Session[idxContext.State] = idxContext` instead of serializing the object?

Reply:
I would have expected yes, but apparently in core the definition for `ISession` doesn't allow for indexing that way.